src/generator/generator_visitor.ts

```diff
@@ -100,42 +103,81 @@ class GeneratorVisitor extends Visitor {
+
+        const operationsInstance = this.getOperationsInstance(commonNumberType);
 
         switch (node.operator) {
             case '+':
-                this.expressions.push(this.module.i32.add(left, right));
+                this.expressions.push(operationsInstance.add(left, right));
                 break;
             case '-':
-                this.expressions.push(this.module.i32.sub(left, right));
+                this.expressions.push(operationsInstance.sub(left, right));
                 break;
             case '*':
-                this.expressions.push(this.module.i32.mul(left, right));
+                this.expressions.push(operationsInstance.mul(left, right));
                 break;
             case '/':
-                this.expressions.push(this.module.i32.div_s(left, right));
+                if (commonNumberType === WebAssemblyType.INT_32) {
+                    this.expressions.push(this.module.i32.div_s(left, right));
+                } else {
+                    this.expressions.push(this.module.f64.div(left, right));
+                }
+
                 break;
             case '%':
-                this.expressions.push(this.module.i32.rem_s(left, right));
+                if (commonNumberType === WebAssemblyType.INT_32) {
+                    this.expressions.push(this.module.i32.rem_s(left, right));
+                } else {
+                    throw new Error('Modulo is not allowed with float values');
+                }
+
                 break;
             case '==':
-                this.expressions.push(this.module.i32.eq(left, right));
+                this.expressions.push(operationsInstance.eq(left, right));
                 break;
             case '!=':
-                this.expressions.push(this.module.i32.ne(left, right));
+                this.expressions.push(operationsInstance.ne(left, right));
                 break;
             case '<':
-                this.expressions.push(this.module.i32.lt_s(left, right));
+                if (commonNumberType === WebAssemblyType.INT_32) {
+                    this.expressions.push(this.module.i32.lt_s(left, right));
+                } else {
+                    this.expressions.push(this.module.f64.lt(left, right));
+                }
+
                 break;
             case '<=':
-                this.expressions.push(this.module.i32.le_s(left, right));
+                if (commonNumberType === WebAssemblyType.INT_32) {
+                    this.expressions.push(this.module.i32.le_s(left, right));
+                } else {
+                    this.expressions.push(this.module.f64.le(left, right));
+                }
+
                 break;
             case '>':
-                this.expressions.push(this.module.i32.gt_s(left, right));
+                if (commonNumberType === WebAssemblyType.INT_32) {
+                    this.expressions.push(this.module.i32.gt_s(left, right));
+                } else {
```

Comment:
Replace with an else if and throw an error in case it's neither `INT_32` nor `FLOAT_64`.